File: Tema 06 - React/petshelter.ui/src/Models/FundraiserInformation.ts
import type { Person } from "./Person";

export class FundraiserInformation {

    name: string;
    endDate: Date;
    target: number;
    status: string;
    current_Amount: number;
    Donors?: Person[];

    constructor(name: string, endDate: Date, target: number, status: string, current_Amount: number, Donors: Person[]) {
        this.name = name;
        this.endDate = endDate;
        this.target = target;
        this.status = status;
        this.current_Amount = current_Amount;
        this.Donors = Donors;
    }
}

File: Tema 07  - Clean code/Before/PetShelter.Domain/FundraiserInformation.cs
namespace PetShelter.Domain
{
    public class FundraiserInformation
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public IReadOnlyCollection<Person> Donors { get; set; }
        public int AmountDonated { get; set; }
        public int Target { get; set; }
        public string Status { get; set; }
        public DateTime EndDate { get; set; }
    }
}
